docs/adr/0003-storage-use-tracking.md

```diff
@@ -0,0 +1,101 @@
+* Does not scale and is not fast. Filesystem walk speed is bottlenecked by IO system calls and the resulting networked calls from the Lustre driver. Using multiple cores will not help speed up processing time significantly.
+* Accesses/uses between filesystem walks will not be recorded.
+
+### Process [Lustre MDS](https://github.com/DDNStorage/lustre_manual_markdown/blob/master/03.01-Monitoring%20a%20Lustre%20File%20System.md#lfs-changelog) changelogs
+
+As of Lustre 2.0 the MDS has a changelog feature that can record [all operations](https://github.com/DDNStorage/lustre_manual_markdown/blob/master/03.01-Monitoring%20a%20Lustre%20File%20System.md#lustre-changelogs) that happen in a Lustre filesystem.
+Note that access time modifications and open events are not recorded by default, but can be recorded by setting the changelog mask:
+
+```shell
+> # lctl set_param mdd.lustre-MDT0000.changelog_mask=OPEN ATIME CREAT UNLINK
+```
+
+The changelog is a simple fixed size ring buffer that will begin to purge old records as the storage fills.
+However, once a changelog reader has been registered Lustre will stop purging records that have not been marked as seen by all readers.
+
+Every record stored contains the following information for every event:
+```
+operation_type(numerical/text) 
+timestamp 
+datestamp 
+flags 
+t=target_FID 
+ef=extended_flags
+u=uid:gid
+nid=client_NID
+p=parent_FID 
+target_name
+```
+
+Caveat: `target_name` refers to the _basename_ of the file being accessed, retreiving the full path requires a lookup on `parent_FID`.
+
+#### Positive Consequences
+
+* Can scale across multiple nodes and cores.
+* Translating open/create/delete events into a reference counting system used for a cache is straight forward.
+* The only system calls made are to resolve parent pathnames, which can be cached. Filesystem driver overhead is minimal.
+* Scaling processing across multiple cores is trivial by assigning each a batch size.
+
+#### Negative Consequences
+
+* Custom data format, requires writing a parser for the CLI output.
+* Very few people seem to interact with Lustre directly at this level, making documentation scarce.
+* When cache state is lost the naive filesystem walk described in the first approach is required to build a view of the filesystem. This will generate Lustre changelog records of its own, a mechanism to filter those out is needed (can be UID/GID).
+
+### Record filesystem changes with `inotify`
+
+`inotify` is a Linux kernel mechanism for recording filesystem events.
+It is only capable of recording events for local VFS devices (i.e. you will only get events for the machine you're using `inotify` on) and is inapplicable here.
+
+## Additional Consideration
+
+### Priority Assets
+
+It may be desirable to provide a set of assets that should never or rarely be evicted, regardless of time or frequency of access.
```

Comment:
Could we arrange the file system as

/a/<files>
and 
/b/<files>

where the process only evicts from /a/ but orchestrator can orchestrate to both, deciding which to orchestrate to based on additional metadata?

Or, we just have another lustre volume that our scavenging process does not run on. Both volumes are mounted by the image servers and when orchestrator proxies it sends the location to the image server as now.